tests/integration/reconcile_test.go

```diff
@@ -83,6 +83,65 @@ func (s *limitAvailModeChangeSuite) TestAvailModeUnchanged() {
 	require.Contains(smbShare.Annotations[backendAnnotation], s.expectBackend)
 }
 
+type scaleoutClusterSuite struct {
+	suite.Suite
+
+	fileSources      []kube.FileSource
+	smbShareResource types.NamespacedName
+
+	// cached values
+	tc *kube.TestClient
+}
+
+func (s *scaleoutClusterSuite) SetupSuite() {
+	// ensure the smbclient test pod exists
+	require := s.Require()
+	s.tc = kube.NewTestClient("")
+	createFromFiles(require, s.tc, s.fileSources)
+	require.NoError(waitForPodExist(s), "smb server pod does not exist")
+	require.NoError(waitForPodReady(s), "smb server pod is not ready")
+}
+
+func (s *scaleoutClusterSuite) TearDownSuite() {
+	deleteFromFiles(s.Require(), s.tc, s.fileSources)
+}
+
+func (s *scaleoutClusterSuite) getTestClient() *kube.TestClient {
+	return s.tc
+}
+
+func (s *scaleoutClusterSuite) getPodFetchOptions() kube.PodFetchOptions {
+	l := fmt.Sprintf(
+		"samba-operator.samba.org/service=%s", s.smbShareResource.Name)
+	return kube.PodFetchOptions{
+		Namespace:     s.smbShareResource.Namespace,
+		LabelSelector: l,
+		MaxFound:      3,
+	}
+}
+
+func (s *scaleoutClusterSuite) TestScaleoutClusterSuite() {
+	require := s.Require()
+	smbShare := &sambaoperatorv1alpha1.SmbShare{}
+	err := s.tc.TypedObjectClient().Get(
+		context.TODO(), s.smbShareResource, smbShare)
+	require.NoError(err)
+
+	// Increase Cluster Size by 1 and check result
+	newClusterSize := smbShare.Spec.Scaling.MinClusterSize + 1
+	smbShare.Spec.Scaling.MinClusterSize = newClusterSize
+	err = s.tc.TypedObjectClient().Update(
+		context.TODO(), smbShare)
+	require.NoError(err)
+	require.NoError(waitForPodExist(s), "smb server pod does not exist")
+	require.NoError(waitForPodReady(s), "smb server pod is not ready")
+	err = s.tc.TypedObjectClient().Get(
+		context.TODO(), s.smbShareResource, smbShare)
+	require.NoError(err)
+	require.Equal(newClusterSize, smbShare.Spec.Scaling.MinClusterSize,
```

Comment:
This comparison is OK to have, but it's not testing that the MinClusterSize value from the smbShare is being reflected in the StatefulSet. We'd need at least another "clause" in the test to fetch the matching StatefulSet and then do something like:
```go
require.Equal(newClusterSize, statefulSet.Spec.Replicas)
```
```

Reply:
Fixed in the latest set of  commit.